http/src/v0/refs.rs

```diff
@@ -0,0 +1,76 @@
+use futures::stream;
+use ipfs::error::Error;
+use ipfs::{Ipfs, IpfsTypes};
+use serde::Serialize;
+use warp::hyper::Body;
+use warp::{path, Filter, Rejection, Reply};
+
+use super::support::with_ipfs;
+
+#[derive(Serialize, Debug)]
+struct RefsResponseItem {
+    #[serde(rename = "Err")]
+    err: String,
+
+    #[serde(rename = "Ref")]
+    refs: String,
+}
+
+/// Handling of https://docs-beta.ipfs.io/reference/http/api/#api-v0-refs-local
+pub fn local<T: IpfsTypes>(
+    ipfs: &Ipfs<T>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    path!("refs" / "local")
+        .and(with_ipfs(ipfs))
+        .and_then(inner_local)
+}
+
+async fn inner_local<T: IpfsTypes>(ipfs: Ipfs<T>) -> Result<impl Reply, Rejection> {
+    let refs: Vec<Result<String, Error>> = ipfs
+        .refs_local()
+        .await
+        .unwrap()
```

Comment:
This could be replaced with `.map_err(StringError::from)?` which would make a json response out of the error, and allow you to continue this chain with the value inside `Ok(_)`.